linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -96,7 +96,7 @@ final class Emitter(config: Emitter.Config, prePrinter: Emitter.PrePrinter) {
   def emit(moduleSet: ModuleSet, logger: Logger): Result = {
     val WithGlobals(body, globalRefs) = emitInternal(moduleSet, logger)
 
-    val result = moduleKind match {
+    val result = coreSpec.moduleKind match {
```

Comment:
Consider adding `import coreSpec._` under `import config._`, like you did in other files.